### PR TITLE
feat: add `sortOnlyDimensions` support for hidden pivot-column sort dims

### PR DESCRIPTION
**Pull request by pimmey**
Related to: PROD-2108

### Description:

Adds support for **sort-only pivot-column dimensions** — hidden pivot dimensions that drive column sort order without appearing as visible pivot column headers.

Previously, all dimensions listed in `pivotConfig.columns` were treated as visible pivot column headers (`groupByColumns`). This change introduces a new `sortOnlyDimensions` field on `PivotConfiguration` to handle the case where a user hides a pivot-column dimension (e.g. `orders_status_priority`) but still wants it to control the order in which pivot columns appear.

**How it works:**

- A pivot-column dimension that is marked hidden (`visible: false`) and has a sort entry is routed to `sortOnlyDimensions` instead of `groupByColumns` in `derivePivotConfigurationFromChart`.
- Hidden pivot-column dims with no sort entry are dropped entirely (no change from before for that case).
- In `PivotQueryBuilder`, `sortOnlyDimensions` are included in the `group_by_query` SELECT and GROUP BY so their values survive the aggregation pipeline, but they are **not** included in the `column_ranking` DISTINCT SELECT — preventing them from creating extra pivot column header groups.
- The `column_ranking` ORDER BY incorporates `sortOnlyDimensions` so they influence which column combination receives `col_idx = 1, 2, 3, …`, producing the desired column order.
- `column_ranking` is now emitted whenever a `sortOnlyDimension` has a matching sort entry (`hasDimSort`), not only when a metric sort is present.
- `convertSqlPivotedRowsToPivotData` requires no changes since `groupByColumns` already contains only visible pivot-column dims by the time it is called.

### CODE REVIEW

**Comment by pimmey** [comment]
* **#23343** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/23343?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lightdash/lightdash/23343?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#23303** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/23303?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23301** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/23301?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23291** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/23291?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23289** <a href="https://app.graphite.com/github/pr/lightdash/lightdash/23289?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/backend/src/utils/QueryBuilder/PivotQueryBuilder.test.ts [hunk 1]
```diff
@@ -5171,4 +5171,202 @@ SELECT * FROM group_by_query LIMIT 50`);
...
+                pivotConfiguration,
+                mockWarehouseSqlBuilder,
+            );
+            const result = builder.toSql();
+
+            // Standard metric sort anchor CTEs must still be emitted
+            expect(result).toContain('"revenue_ca" AS (');
+            expect(result).toContain('column_ranking AS (');
+        });
+    });
```
[the remark this answers is not in the document]
Comment (pimmey) [reply packages/backend/src/utils/QueryBuilder/PivotQueryBuilder.test.ts:5374]: Pushing back on this one — `PivotQueryBuilder.test.ts` is a unit test suite for an SQL builder, and asserting on generated SQL strings is the established pattern across the entire 5000+ LOC file. Examples are everywhere (`expect(result).toContain(...)` with `replaceWhitespace` helpers). For a class whose single responsibility is producing SQL, the *generated SQL string is the business logic outcome* — testing it isn't "testing implementation details," it's testing the behavior. The bot's suggested `builder.getPivotColumns()` API doesn't exist.

That said, the spec reviewer did flag two real things worth fixing in the new tests (a misleading description on the no-sort defensive guard, and a missing edge case for all-hidden-pivot-dims). I'll address those separately. Marking this thread resolved.

**Comment by pimmey** [comment]
### Merge activity

* **May 20, 5:55 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/lightdash/lightdash/23343).
* **May 20, 6:01 PM UTC**: [Graphite](https://app.graphite.com/github/pr/lightdash/lightdash/23343) rebased this pull request as part of a merge.
